File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        return depthHelper(root);
    }

    private int depthHelper(TreeNode curr) {
        if (curr == null) {
            return 0;
        }
        int leftDepth = depthHelper(curr.left);
        int rightDepth = depthHelper(curr.right);
        return Math.max(leftDepth, rightDepth) + 1;
    }

}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        if not root:
            return 

        val = root.val
        if p.val <= val <= q.val or q.val <= val <= p.val:
            return root
        elif val < p.val:
            root = root.right
        else:
            root = root.left

        return self.lowestCommonAncestor(root, p, q)